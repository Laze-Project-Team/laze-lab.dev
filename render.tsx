import type {
  Queries,
  queries,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { render as defaultRender } from '@testing-library/react';
import type { FC, ReactNode } from 'react';

import { ColorModeProvider } from '@/components/contexts/ColorModeContext';

export type renderType<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container,
> = (
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>,
) => RenderResult<Q, Container, BaseElement>;

type wrapperProps = {
  children?: ReactNode;
};

type globalWrapperHOCType = (
  wrapper?: React.JSXElementConstructor<{ children: React.ReactElement }>,
) => FC<wrapperProps>;

const globalWrapperHOC: globalWrapperHOCType = (Wrapper) => {
  const globalWrapper: FC<wrapperProps> = ({ children }) => (
    <ColorModeProvider>
      {Wrapper ? (
        <Wrapper>
          <>{children}</>
        </Wrapper>
      ) : (
        children
      )}
    </ColorModeProvider>
  );

  return globalWrapper;
};

export const render: renderType = (ui, options) =>
  defaultRender(ui, {
    ...options,
    wrapper: globalWrapperHOC(options?.wrapper),
  });
