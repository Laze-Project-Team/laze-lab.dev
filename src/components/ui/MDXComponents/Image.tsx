import { css } from '@emotion/react';
import NextImage from 'next/image';
import type { DetailedHTMLProps, FC, ImgHTMLAttributes } from 'react';

export const Image: FC<
  DetailedHTMLProps<ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>
> = ({ src, placeholder: _, alt, width, height }) => (
  <div
    css={css`
      width: 100%;
      max-width: 40rem;
    `}
  >
    {src && (
      <NextImage
        src={src}
        fill
        css={css`
          object-fit: contain;
        `}
        alt={alt ?? ''}
        width={Number(width)}
        height={Number(height)}
      />
    )}
  </div>
);
