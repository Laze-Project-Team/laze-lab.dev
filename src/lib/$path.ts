export const pagesPath = {
  "login": {
    $url: (url?: { hash?: string }) => ({ pathname: '/login' as const, hash: url?.hash })
  },
  "signup": {
    $url: (url?: { hash?: string }) => ({ pathname: '/signup' as const, hash: url?.hash })
  },
  $url: (url?: { hash?: string }) => ({ pathname: '/' as const, hash: url?.hash })
}

export type PagesPath = typeof pagesPath

export const staticPath = {
  favicon_ico: '/favicon.ico',
  favicons: {
    android_chrome_192x192_png: '/favicons/android-chrome-192x192.png',
    android_chrome_384x384_png: '/favicons/android-chrome-384x384.png',
    apple_touch_icon_png: '/favicons/apple-touch-icon.png',
    browserconfig_xml: '/favicons/browserconfig.xml',
    favicon_16x16_png: '/favicons/favicon-16x16.png',
    favicon_32x32_png: '/favicons/favicon-32x32.png',
    favicon_ico: '/favicons/favicon.ico',
    mstile_150x150_png: '/favicons/mstile-150x150.png',
    safari_pinned_tab_svg: '/favicons/safari-pinned-tab.svg',
    site_webmanifest: '/favicons/site.webmanifest'
  },
  img: {
    logo: {
      logo_png: '/img/logo/logo.png',
      logo_caption_png: '/img/logo/logo_caption.png',
      logo_caption_gray_png: '/img/logo/logo_caption_gray.png',
      logo_gray_png: '/img/logo/logo_gray.png'
    }
  },
  vercel_svg: '/vercel.svg'
} as const

export type StaticPath = typeof staticPath
