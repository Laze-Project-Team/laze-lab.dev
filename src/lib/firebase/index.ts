import { getAnalytics, isSupported } from 'firebase/analytics';
import type { FirebaseOptions } from 'firebase/app';
import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import { getPerformance } from 'firebase/performance';

const firebaseConfig: FirebaseOptions = {
  apiKey: 'AIzaSyAZSNEXg1lQEP_5OufUbJ_wpAxMTcevmio',
  authDomain: 'laze-d339e.firebaseapp.com',
  projectId: 'laze-d339e',
  storageBucket: 'laze-d339e.appspot.com',
  messagingSenderId: '280279659559',
  appId: '1:280279659559:web:a454d4625f5c92fa5a4a02',
  measurementId: 'G-EF2HM3ELZ8',
};

export const app = initializeApp(firebaseConfig);

// Analytics
export const analytics = isSupported().then((supported) => {
  return supported ? getAnalytics(app) : null;
});

// Performance
export const perf = getPerformance(app);

// Authentication
export const auth = getAuth();
