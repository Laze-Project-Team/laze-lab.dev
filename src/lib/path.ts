export const rootDir = process.cwd();

export const contentsDir = `${rootDir}/src/contents`;

export const pagesDir = `${rootDir}/src/pages`;
