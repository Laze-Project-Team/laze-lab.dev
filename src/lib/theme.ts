import { createTheme } from '@material-ui/core';

export const theme = createTheme({
  overrides: {
    MuiAppBar: {
      colorDefault: {
        backgroundColor: 'var(--color-darkgray)',
      },
    },
  },
});
