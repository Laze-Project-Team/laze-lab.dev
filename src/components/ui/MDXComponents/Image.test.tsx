import '@testing-library/jest-dom';

import { expect, it } from 'vitest';

import { staticPath } from '@/lib/$path';
import { render } from '@/lib/test/render';

import { Image } from './Image';

it('should be rendered correctly', () => {
  const { container } = render(
    <Image src={staticPath.img.logo.logo_png} alt="logo" />,
  );
  expect(container.firstChild).toMatchSnapshot();
});
