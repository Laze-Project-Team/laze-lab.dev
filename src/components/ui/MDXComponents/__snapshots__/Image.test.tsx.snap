// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should be rendered correctly 1`] = `
<div
  class="css-1o4q9st-Image"
>
  <span
    style="box-sizing: border-box; display: block; overflow: hidden; width: initial; height: initial; background: none; opacity: 1; border: 0; margin: 0; padding: 0; position: absolute; top: 0; left: 0; bottom: 0; right: 0;"
  >
    <img
      alt="logo"
      data-nimg="fill"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0; left: 0; bottom: 0; right: 0; box-sizing: border-box; padding: 0; border: none; margin: auto; display: block; width: 0; height: 0; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
    />
    <noscript />
  </span>
</div>
`;
